src/EFTools/EntityDesignerVersioningFacade/ReverseEngineerDb/SchemaDiscovery/EntityStoreSchemaGeneratorDatabaseSchemaLoader.cs

```diff
@@ -15,17 +18,44 @@ namespace Microsoft.Data.Entity.Design.VersioningFacade.ReverseEngineerDb.Schema
     /// </summary>
     internal class EntityStoreSchemaGeneratorDatabaseSchemaLoader
     {
+        private const string SwitchOffMetadataMergeJoins = "Switch.Microsoft.Data.Entity.Design.DoNotUseSqlServerMetadataMergeJoins";
         private readonly EntityConnection _connection;
         private readonly Version _storeSchemaModelVersion;
+        private readonly IDbCommandInterceptor _addOptionMergeJoinInterceptor;
 
-        public EntityStoreSchemaGeneratorDatabaseSchemaLoader(EntityConnection entityConnection, Version storeSchemaModelVersion)
+        [SuppressMessage("Microsoft.Usage", "CA1806:DoNotIgnoreMethodResults", MessageId = "System.Boolean.TryParse(System.String,System.Boolean@)")]
+        public EntityStoreSchemaGeneratorDatabaseSchemaLoader(
+            EntityConnection entityConnection,
+            Version storeSchemaModelVersion,
+            bool needsMergeJoinHint)
         {
             Debug.Assert(entityConnection != null, "entityConnection != null");
             Debug.Assert(entityConnection.State == ConnectionState.Closed, "expected closed connection");
             Debug.Assert(EntityFrameworkVersion.IsValidVersion(storeSchemaModelVersion), "invalid version");
 
             _connection = entityConnection;
             _storeSchemaModelVersion = storeSchemaModelVersion;
+
+            if (needsMergeJoinHint)
+            {
```

Comment:
can we extract all the quirk logic into its own private method for clarity?

Reply:
Done.